web/js/helpers/OTHelper.js

```diff
@@ -257,10 +257,10 @@
       if (!Array.isArray(aHandlers)) {
         aHandlers = [aHandlers];
       }
-      return otLoaded.then(function() {
-        return new Promise(function(resolve, reject) {
+      return new Promise(function(resolve, reject) {
```

Comment:
I like this pattern with the new Promise and an explicit reject or resolve later on in the body. 
We could try and ensure we follow this is other projects.